modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts

```diff
@@ -0,0 +1,288 @@
+      const userKeychain = await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase',
+        'originalPasscodeEncryptionCode');
+      const backupKeychain = await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+      userKeychain.should.deepEqual(nockedUserKeychain);
+
+      // unencrypted `prv` property should exist on backup keychain
+      const keyChainPrv = JSON.parse(backupKeychain.prv ?? '');
+      _.isEqual(keyChainPrv.pShare, backupKeyShare.pShare).should.be.true();
+      _.isEqual(keyChainPrv.bitgoNShare, bitgoKeyShare.nShares[2]).should.be.true();
+      _.isEqual(keyChainPrv.userNShare, userKeyShare.nShares[2]).should.be.true();
+      should.exist(backupKeychain.encryptedPrv);
+    });
+
+    it('should fail to generate TSS key chains', async function () {
+      const nockedBitGoKeychain = await nockBitgoKeychain({
+        coin: coinName,
+        userKeyShare,
+        backupKeyShare,
+        userGpgKey,
+      });
+      const bitgoKeychain = await tssUtils.createBitgoKeychain(userGpgKey, userKeyShare, backupKeyShare);
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+
+      await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        await MPC.keyShare(2, 2, 3),
+        bitgoKeychain,
+        'passphrase').should.be.rejectedWith('Common keychains do not match');
+      await tssUtils.createUserKeychain(
+        userGpgKey,
+        await MPC.keyShare(1, 2, 3),
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase')
+        .should.be.rejectedWith('Common keychains do not match');
+
+      const share = await MPC.keyShare(1, 2, 3);
+      await tssUtils.createBackupKeychain(
+        userGpgKey,
+        share,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase')
+        .should.be.rejectedWith('Common keychains do not match');
+      await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        await MPC.keyShare(2, 2, 3),
+        bitgoKeychain,
+        'passphrase')
+        .should.be.rejectedWith('Common keychains do not match');
+    });
+  });
+
+  // #region Nock helpers
+  async function nockBitgoKeychain(params: {
+    coin: string,
+    userKeyShare: KeyShare,
+    backupKeyShare: KeyShare,
+    userGpgKey: openpgp.SerializedKeyPair<string>,
+  }): Promise<Keychain> {
+    const bitgoCombined = MPC.keyCombine(bitgoKeyShare.pShare, [params.userKeyShare.nShares[3], params.backupKeyShare.nShares[3]]);
+    const userToBitgoShare = await encryptNShare(
+      bitgoKeyShare,
+      1,
+      userGpgKey.publicKey,
+      userGpgKey.privateKey
+    );
+    const backupToBitgoShare = await encryptNShare(
+      bitgoKeyShare,
+      2,
+      userGpgKey.publicKey,
```

Comment:
Lets may be use an independent gpg key pair dedicated for backup since user may not always be the one to generate backup keys